torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -3105,6 +3141,12 @@ def test_all_to_all(self):
             group, group_id, rank = self._init_global_test()
             self._test_all_to_all_helper(group, group_id, rank)
 
+        @sandcastle_skip_if(BACKEND != "nccl", "Only NCCL supports all_to_all")
+        def test_all_to_all_quantized(self):
```

Comment:
There're some test failures related to this, see https://app.circleci.com/pipelines/github/pytorch/pytorch/359751/workflows/b2601889-d460-4429-abd9-882385465852/jobs/15174372 and https://app.circleci.com/pipelines/github/pytorch/pytorch/359751/workflows/b2601889-d460-4429-abd9-882385465852/jobs/15174373

It seems like if you want to use nccl backend only, you need to pass in cuda tensors right? otherwise nccl will not accept it. You probably want to use `cuda` and `rank_to_GPU` argument if you want to use the all_to_all_helper

Reply:
Yeah, I'm currently adding them. Also, using GPU servers to test them internally.